packages/meditrak-server/src/routes/GETAnswers/assertAnswerPermissions.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { QUERY_CONJUNCTIONS, TYPES } from '@tupaia/database';
+import { hasBESAdminAccess } from '../../permissions';
+import {
+  assertSurveyResponsePermissions,
+  createSurveyResponseDBFilter,
+} from '../GETSurveyResponses';
+
+const { RAW } = QUERY_CONJUNCTIONS;
+
+export const assertAnswerPermissions = async (accessPolicy, models, answerId) => {
+  const answer = await models.answer.findById(answerId);
+  if (!answer) {
+    throw new Error(`No answer exists with id ${answerId}`);
+  }
+
+  await assertSurveyResponsePermissions(accessPolicy, models, answer.survey_response_id);
+
+  return true;
+};
+
+export const createAnswerDBFilter = async (accessPolicy, models, criteria, options) => {
+  const dbConditions = { ...criteria };
+  const dbOptions = { ...options };
+
+  if (hasBESAdminAccess(accessPolicy)) {
+    return { dbConditions, dbOptions };
+  }
+
+  const allPermissionGroupsNames = accessPolicy.getPermissionGroups();
+  const countryCodesByPermissionGroupId = {};
+
+  // Generate lists of country codes we have access to per permission group id
+  for (const permissionGroupName of allPermissionGroupsNames) {
+    const permissionGroup = await models.permissionGroup.findOne({ name: permissionGroupName });
+    if (permissionGroup) {
+      const countryNames = accessPolicy.getEntitiesAllowed(permissionGroupName);
+      countryCodesByPermissionGroupId[permissionGroup.id] = countryNames;
+    }
+  }
+
+  // Join SQL table with survey_response, entity and survey tables
+  // Running the permissions filtering is much faster with joins
+  dbOptions.multiJoin = [
```

Comment:
An edge case here is that a different `multiJoin` is specified by the incoming `options`, and we overwrite it.